### ISSUE TITLE
[Feature request] Support `SAME` padding for convolutional layers

### ISSUE THREAD

**Issue by zuoxingdong** [pytorch/pytorch#7549]
It would be very convenient to have `SAME` padding supported for convolutional layers

**Comment by zou3519** [comment]
What is "same" padding? Could you give an example?

**Comment by zuoxingdong** [comment]
@zou3519 For example, it could be easier to have the output size identical to input size if the stride is 1. More generally, 

```
out_H = torch.ceil(in_H/stride)
out_W = torch.ceil(in_W/stride)
```

**Comment by crcrpar** [comment]
"SAME" padding option is used in TensorFlow. It's explained in below document.
https://www.tensorflow.org/api_guides/python/nn#Convolution

**Comment by SsnL** [comment]
I'm personally not a fan of `"same"` as a padding option. It hides some degrees of freedom from user, e.g. when kernel size is even.

**Comment by fmassa** [comment]
Duplicate of https://github.com/pytorch/pytorch/issues/3867
I personally feel that we should not support this except if the behavior exactly matches tensorflow.
But for that, we would need to have asymmetric padding for our operations, which would involve some non-trivial changes. I've discussed those points in more detail in #3867 

Closing as a duplicate.
